packages/smooth_app/lib/cache/files/files_cache_impl.dart

```diff
@@ -0,0 +1,129 @@
+// ignore_for_file: avoid_slow_async_io
+part of 'files_cache.dart';
+
+/// Implementation based on [getTemporaryDirectory]
+class _ShortLivingFileCacheManagerImpl extends _FileCacheManagerImpl {
+  _ShortLivingFileCacheManagerImpl({
+    required String subFolderName,
+  }) : super._(
+          rootDirectory: getTemporaryDirectory(),
+          subFolderName: subFolderName,
+        );
+}
+
+/// Implementation based on [getApplicationDocumentsDirectory]
+class _LongLivingFilesCacheManagerImpl extends _FileCacheManagerImpl {
+  _LongLivingFilesCacheManagerImpl({
+    required String subFolderName,
+  }) : super._(
+          rootDirectory: getApplicationDocumentsDirectory(),
+          subFolderName: subFolderName,
+        );
+}
+
+class _FileCacheManagerImpl extends FileCache {
+  _FileCacheManagerImpl._({
+    required this.rootDirectory,
+    required this.subFolderName,
+  }) : assert(subFolderName.isNotEmpty);
+
+  final Future<Directory> rootDirectory;
+  final String subFolderName;
+
+  /// This field will be lazily assigned after a call to [_initCache]
+  Directory? _directory;
+
+  @protected
+  Future<void> _initCache() async {
```

Comment:
My Java roots tell me: if it's `protected` it's not private.
I suggest `initCache()`.